packages/x-charts/src/internals/seriesSelectorOfType.ts

```diff
@@ -0,0 +1,71 @@
+import { warnOnce } from '@mui/x-internals/warning';
+import { createSelector, createSelectorMemoized } from '@mui/x-internals/store';
+import { ChartSeriesDefaultized, ChartsSeriesConfig } from '../models/seriesType/config';
+import { SeriesId } from '../models/seriesType/common';
+import { selectorChartSeriesProcessed } from './plugins/corePlugins/useChartSeries/useChartSeries.selectors';
+import type { ProcessedSeries } from './plugins/corePlugins/useChartSeries';
+import { useStore } from './store/useStore';
+import { useSelector } from './store/useSelector';
+
+export const selectorAllSeriesOfType = createSelector(
+  selectorChartSeriesProcessed,
+  <T extends keyof ChartsSeriesConfig>(processedSeries: ProcessedSeries, seriesType: T) =>
+    processedSeries[seriesType],
+);
+
+export const selectorSeriesOfType = createSelectorMemoized(
+  selectorChartSeriesProcessed,
+  <T extends keyof ChartsSeriesConfig>(
+    processedSeries: ProcessedSeries,
+    seriesType: T,
+    ids?: SeriesId | SeriesId[],
+  ) => {
+    if (!ids || (Array.isArray(ids) && ids.length === 0)) {
+      return (
+        processedSeries[seriesType]?.seriesOrder?.map(
+          (seriesId) => processedSeries[seriesType]?.series[seriesId],
+        ) ?? []
+      );
+    }
+
+    if (!Array.isArray(ids)) {
+      return processedSeries[seriesType]?.series?.[ids];
+    }
+
+    const result: ChartSeriesDefaultized<T>[] = [];
+    const failedIds: SeriesId[] = [];
+    for (const id of ids) {
+      const series = processedSeries[seriesType]?.series?.[id];
+      if (series) {
+        result.push(series);
+      } else {
+        failedIds.push(id);
```

Comment:
If we don't want this to have a performance impact in production environments we can derive the failed IDs from `result` and `ids` inside the if statement. 

Likely not a big deal, though

Reply:
Yeah I would assume this hook is not called with a lot of ids